apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -806,55 +806,67 @@ func (m *ModelManagerAPI) ListModelSummaries(ctx context.Context, req params.Mod
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelSummariesForUser(userTag, req.All && m.isAdmin)
+	var modelInfos []model.ModelSummary
+	if req.All {
+		if !m.isAdmin {
+			return result, fmt.Errorf("%w: cannot list all models as non-admin user", apiservererrors.ErrPerm)
+		}
+		modelInfos, err = m.modelService.ListAllModelSummaries(ctx)
+	} else {
+		modelInfos, err = m.modelService.ListModelSummariesForUser(ctx, userTag.Id())
+	}
 	if err != nil {
 		return result, errors.Trace(err)
 	}
+
 	err = m.fillInStatusBasedOnCloudCredentialValidity(ctx, modelInfos)
```

Comment:
Correct me if I am  wrong but I think we can establish this information in the summary call and reduce our need to go back to the DB twice?

Reply:
I didn't touch this mostly because status has not been moved over yet. Right now we don't fill in status at all in state but I have put todos to add it to the model summery once it is. 

Imo this is a temporary ugly measure that is a good reminder that this needs to be moved into state once its ready. I will put a comment saying as such next to it.

I had thought it used state, it does, but only in a very superficial way.

Reply:
ok